### PR TITLE
Bugfix/dcw questionnaire logic

### PR DESCRIPTION
**Pull request by achen2401**
https://www.pivotaltracker.com/story/show/184364436
https://www.pivotaltracker.com/story/show/184364472
https://www.pivotaltracker.com/story/show/184364457

Changes include:
- Added missing check for LOINC code, G30.9 (Alzheimer's disease, unspecified) for dementia condition (addressed https://www.pivotaltracker.com/story/show/184364436)
- Added dementia condition as a prerequisite for displaying C-IDAS , in addition to low SLUMs score (address https://www.pivotaltracker.com/story/show/184364472)
- Move ADL-IADL questionnaire up in the order of questionnaires presented (before BEHAV5) (address https://www.pivotaltracker.com/story/show/184364457)

Other changes:
- add additional tests for changed logic

### CODE REVIEW

File: src/cql/upstream-external/CirgLibraryQuestionnaireLogic_DCW.cql [hunk 1]
```diff
@@ -384,14 +394,15 @@ define PresentQnr_CIRG_GDS:
...
   if Length(CIRG_C_IDAS_Responses_Today) >= 1 then null
-  // CIRG-SLUMS score indicates dementia 
-  else if dcw_mood_level_2 is true and 
-     dcw_care_partner_present is true and 
-     HasSLUMSDementiaScore is true
+  // dementia diagnosis/condition or CIRG-SLUMS score indicates dementia 
+  else if HasDementia or (
+    dcw_mood_level_2 is true and 
+    dcw_care_partner_present is true and 
+    HasSLUMSDementiaScore is true)
```
Comment (mcjustin) [inline src/cql/upstream-external/CirgLibraryQuestionnaireLogic_DCW.cql:403]: I believe this is incorrect and that the conditions should be:
CP present AND Dementia from SLUMS.
Per [our curated gdoc](https://docs.google.com/document/d/1_6tiOwgPt8xMvLrRR8U234X-31ZhYlK2/edit#bookmark=id.1zczfw5557ti) and confirmed upstream in [sharepoint](https://uwnetid.sharepoint.com/:w:/r/sites/og_son_dpen_team/_layouts/15/Doc.aspx?sourcedoc=%7BD118B699-1D5E-4C0A-8931-422CCE5E7306%7D&file=DCW_8.9.22.docx&action=default&mobileredirect=true).

Looks like the tests will need editing too.

**Review by mcjustin — CHANGES_REQUESTED** [review]

**Comment by mcjustin** [comment]
@achen2401  could you please edit your initial comment here to map the pivotal story to the change? Sorry to ask for that detail, but I had trouble mapping them myself.

**Comment by achen2401** [comment]
thanks @mcjustin I fixed the description of the PR so that each change is mapped to the story.

File: src/cql/upstream-external/CirgLibraryQuestionnaireLogic_DCW.cql (hunk 1 above)
Comment (achen2401) [reply src/cql/upstream-external/CirgLibraryQuestionnaireLogic_DCW.cql:403]: @mcjustin hmm, okay, this change is in response to the story:  https://www.pivotaltracker.com/story/show/184364472  It sounds like a Dementia diagnosis/condition should also trigger administration of SLUMS?

Comment (sierramatice) [reply src/cql/upstream-external/CirgLibraryQuestionnaireLogic_DCW.cql:405]: this looks correct to me! it is SLUMs indicating dementia OR dementia DX

Comment (mcjustin) [reply src/cql/upstream-external/CirgLibraryQuestionnaireLogic_DCW.cql:405]: @sierramatice  I see that you edited the spec - thanks. Copying it here for convenience:
```
Condition: CP gateway question: CP present 
AND
Condition: Dementia from SLUMS OR dementia DX
```
And noting the overall mood/behavior level 2 criteria of:
```
depression or anxiety diagnosed in the EMR 
Behave5+ failed 
```
The proposed change here, however, does not match what you wrote. 
```
  else if HasDementia or (
    dcw_mood_level_2 is true and 
    dcw_care_partner_present is true and 
    HasSLUMSDementiaScore is true)
```
I'm on calls back to back today, but that's my 2 cents...

Comment (sierramatice) [reply src/cql/upstream-external/CirgLibraryQuestionnaireLogic_DCW.cql:405]: Oh! I misinterpreted the code, I focused on the comment `// dementia diagnosis/condition or CIRG-SLUMS score indicates dementia `

@achen2401 the care_partner_present always needs to be true to present this assessment!

**Review by mcjustin — APPROVED** [review]

**Pull request merged**